src/main/java/com/amit/electronic/store/service/UserServiceImpl.java

```diff
@@ -0,0 +1,69 @@
+package com.amit.electronic.store.service;
+
+import com.amit.electronic.store.entity.User;
+import com.amit.electronic.store.model.GetUserRequest;
+import com.amit.electronic.store.repository.UserRepository;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Transactional
+@AllArgsConstructor
+@Slf4j
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+
+    public User create(User user) {
+        return userRepository.save(user);
+    }
+
+    public void deleteUser(User user) {
+        userRepository.delete(user);
+    }
+
+    public List<User> getAllUsers() {
+        return userRepository.findAll();
+    }
+
+    public User setUserEmail(User user, String email) {
+        user.setEmail(email);
+        return userRepository.save(user);
+    }
+
+    public User updateUser(User user) {
+        User updatedUser = userRepository.findById(user.getId()).orElse(null);
+        if (updatedUser != null) {
+            updatedUser.setName(user.getName());
+            updatedUser.setEmail(user.getEmail());
+            updatedUser.setPassword(user.getPassword());
```

Comment:
Can password really be updated like this?

If you're updating password, you need a different endpoint for that.

Reply:
removed